PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.h

```diff
@@ -63,24 +69,159 @@
 #define PM_MOVEUPENTRY             TEXT("Move Up\tCtrl+Up")
 #define PM_MOVEDOWNENTRY           TEXT("Move Down\tCtrl+Down")
 
-enum NodeType {
-	nodeType_root = 0, nodeType_project = 1, nodeType_folder = 2, nodeType_file = 3
-};
-
 class TiXmlNode;
 
-class ProjectPanel : public DockingDlgInterface {
+
+
+
+class ProjectPanel : public DockingDlgInterface, public TreeViewListener {
+
+	friend int CALLBACK compareFunc(LPARAM, LPARAM, LPARAM);
+
+	enum NodeType { nodeType_dummy = -1, nodeType_root = 0, nodeType_project = 1, nodeType_folder = 2, nodeType_file = 3, nodeType_baseDir = 4, nodeType_dir = 5, nodeType_dirFile = 6, };
+
+	// private class ProjectPanelData is stored in the LPARAM of the tree nodes.
+	// The data it contains depends on the node type.
+	class ProjectPanelData final : public TreeViewData {
+	public:
+		generic_string _name;
+		generic_string _filePath;
+		generic_string _userLabel;
+		std::vector<generic_string> _filters;
+		NodeType _nodeType;
+		DirectoryWatcher& _directoryWatcher;
+		HTREEITEM _hItem;
+		bool _watch;
+
+		ProjectPanelData(DirectoryWatcher& directoryWatcher, const TCHAR* name, const TCHAR* filePath, NodeType nodeType, const std::vector<generic_string> filters = std::vector<generic_string>()) 

```

Comment:
The `const TCHAR*` arguments should be one of:

- `_In_z_ const TCHAR*`
- `PCTSTR`